mlflow/genai/datasets/__init__.py

```diff
@@ -0,0 +1,24 @@
+"""
+Databricks Agent Datasets Python SDK. For more details see Databricks Agent Evaluation:
+ <https://docs.databricks.com/en/generative-ai/agent-evaluation/index.html>
+"""
+
+try:
+    from databricks.agents.datasets import (
+        create_dataset,
+        Dataset,
+        delete_dataset,
+        get_dataset,
+    )
+except ImportError:
+    raise ImportError(
+        "The `databricks-agents` package is required to use mlflow.genai.datasets. "
+        "Please install it with `pip install databricks-agents`."
+    )
+
+__all__ = [
+    "create_dataset",
+    "Dataset",
```

Comment:
Shall we call this `EvaluationDataset` to disambiguate it from the other `Dataset` objects in MLflow?

Can we also ensure that this inherits from `mlflow.data.Dataset`?

Reply:
+1

Reply:
Sorry, `mlflow.entities.Dataset` is correct. Thanks!

Reply:
@smoorjani can we update this to inherit from `mlflow.data.Dataset`?

Reply:
@smoorjani Can we inherit from `mlflow.data.Dataset`? It's fine if we define a new class in MLflow that does some wrapping. Inheriting from `mlflow.data.Dataset` is the necessary approach for compatibility with `mlflow.log_inputs()`, etc.

Reply:
Yes, confirmed with Daniel, I'll get a PR out for this